images/helm-operator/run-operator.sh

```diff
@@ -259,38 +261,45 @@ while true; do
             HELM_ARGS+=("-f" "$EXTRA_VALUES_FILE")
         fi
 
-        if [ -z "$RESOURCE_VALUES" ]; then
-            echo "No values for $FULL_RESOURCE_NAME, using default values"
-        else
-            VALUES_FILE="/tmp/$RESOURCE_NAMESPACE-$RESOURCE_NAME-values.yaml"
-            echo -E "$RESOURCE_VALUES" > "$VALUES_FILE"
+        VALUES_FILE="/tmp/$RESOURCE_NAMESPACE-$RESOURCE_NAME-values.yaml"
+        NEW_VALUES_FILE="/tmp/$RESOURCE_NAMESPACE-$RESOURCE_NAME-values-NEW.yaml"
 
-            HELM_ARGS+=("-f" "$VALUES_FILE")
-        fi
+        echo -E "$RESOURCE_VALUES" > "$NEW_VALUES_FILE"
 
-        mkdir -p "$RESOURCE_VERSIONS_DIRECTORY"
-        RESOURCE_VERSION_FILE="$RESOURCE_VERSIONS_DIRECTORY/$RESOURCE_NAMESPACE-$RESOURCE_NAME.resourceVersion"
-        # If the resource version for this Release CR hasn't changed, we can skip running helm upgrade.
-        if [[ -s "$RESOURCE_VERSION_FILE" && "$(cat "$RESOURCE_VERSION_FILE")" == "$RESOURCE_RESOURCE_VERSION" ]]; then
+        # If the spec for this CR hasn't changed, we can skip running helm upgrade.
+        if [[ -a "$VALUES_FILE" && "$(cat "$VALUES_FILE")" == "$(cat "$NEW_VALUES_FILE")" ]]; then
             echo "Nothing has changed for $FULL_RESOURCE_NAME"
         else
-            echo "$CRD $FULL_RESOURCE_NAME has been modified"
+            if [ -a "$VALUES_FILE" ]; then
+                echo "$CRD $FULL_RESOURCE_NAME has been modified"
+            else
+                echo "New $CRD $FULL_RESOURCE_NAME"
+            fi
+            HELM_ARGS+=("-f" "$NEW_VALUES_FILE")
 
             writeReleaseOwnerValuesFile "$RESOURCE_API_VERSION" "$RESOURCE_KIND" "$RESOURCE_NAME" "$RESOURCE_UID"
             writeReleaseConfigMapOwnerPatchFile "$RESOURCE_API_VERSION" "$RESOURCE_KIND" "$RESOURCE_NAME" "$RESOURCE_UID"
             HELM_ARGS+=("-f" "$OWNER_VALUES_FILE")
 
             mkdir -p "$UPGRADE_RESULT_DIRECTORY"
             echo "Running helm upgrade for release $RELEASE_NAME"
-            if helmUpgrade "$RELEASE_NAME" "$CHART" "$RESOURCE_NAMESPACE" "${HELM_ARGS[@]}" | tee "$UPGRADE_RESULT_DIRECTORY/$RELEASE_NAME.txt"; then
-                echo "Error occurred when processing $FULL_RESOURCE_NAME"
+            if helmUpgrade "$RELEASE_NAME" "$CHART" "$RESOURCE_NAMESPACE" "${HELM_ARGS[@]}"; then
```

Comment:
The if conditions here were backwards, so it always errored. I don't know how I managed that.